zero/gkyl_position_map_priv.h

```diff
@@ -0,0 +1,448 @@
+#include <gkyl_array.h>
+#include <gkyl_calc_bmag.h>
+
+struct opt_Theta_ctx
+{
+  struct gkyl_position_map *gpm;
+  struct gkyl_bmag_ctx *bmag_ctx;
+  double dB_target;
+  double dB_global_lower;
+  double B_lower_region;
+};
+
+// Allocate double array (filled with zeros).
+static struct gkyl_array*
+mkarr(bool on_gpu, long nc, long size)
+{
+  struct gkyl_array* a;
+  a = gkyl_array_new(GKYL_DOUBLE, nc, size);
+  return a;
+}
+
+/**
+ * Function that actually frees memory associated with this
+ * object when the number of references has decreased to zero.
+ *
+ * @param ref Reference counter for this object.
+ */
+void gkyl_position_map_free(const struct gkyl_ref_count *ref);
+
+// Utility functions for polynomial based B mapping
+
+void
+calculate_mirror_throat_location_polynomial(struct gkyl_position_map_const_B_ctx *constB_ctx, struct gkyl_bmag_ctx *bmag_ctx)
+{
+  // Assumes symmetry along theta centered at 0 and two local maxima in Bmag that are symmetric
+
+  enum { X_IDX, Y_IDX, Z_IDX }; // arrangement of cartesian coordinates
+  double psi = constB_ctx->psi;
+  double alpha = constB_ctx->alpha;
+  double *xp = malloc(3*sizeof(double));
```

Comment:
here and line 49 below, please don't use malloc for small arrays whose size is known at compile time. Just use double xp[3].